### ISSUE TITLE
make privacy mitigations mandatory

### ISSUE THREAD

**Issue by samuelweiler** [w3c/wot-thing-description#794]
During the Privacy IG (PING) call on 20 June 2019, ([minutes](https://www.w3.org/Privacy/IG/summaries/PING-minutes-20190620)) representatives from the WG indicated that the non-normative (and pretty non-specific) privacy and security guidance in section 9 would be clarified in future documents. [PING's view](https://www.w3.org/blog/2019/06/privacy-anti-patterns-in-standards/) is that privacy "[p]rotections need to be specific, as mandatory as the defined behavior, and defined from day one ..."

For example: on the call we observed that the "id" field could have some privacy harms.  It is not sufficient to say "this will be addressed in a future doc".  If the privacy mitigations are not cooked enough to be published yet, then the scheme as a whole is not cooked enough to be published yet.

**Comment by snyderp** [comment]
Just to second what @samuelweiler said above, PING is having a difficult time evaluating the spec.

This is mostly because the privacy implications of whats described currently depend on the interaction of the data model and the yet-to-be-specified APIs; w/o seeing the forth-coming APIs, and considering both halves in tandem, we can't evaluate the privacy impact.

We're looking forward to giving the entire system a review when its ready though, and would be very happy and eager to work with you while the API parts are being spec'ed out, to ensure its done in a privacy-preserving manner (and to make the process as smooth as possible on your end)

**Comment by samuelweiler** [comment]
@plehegar

**Comment by mmccool** [comment]
The TD spec itself is a data format whose purpose is to encode and provide metadata.  We agree that some of this metadata is privacy sensitive but we can't remove it without hurting the intended functionality.  So we do have to seek ways to protect privacy via specifying suitable distribution mechanisms.

To clarify, by "API" you mean the (proposed) Thing Directory API used for discovering, accessing and managing the distribution of TDs, correct?   This is a topic we are planning for our next WG charter.  We would be happy to work with you throughout the process of developing such a specification to ensure it supports the preservation of privacy.

**Comment by snyderp** [comment]
@mmccool This kind of sentence is exactly the thing that gets PING worried :) 

> We agree that some of this metadata is privacy sensitive but we can't remove it without hurting the intended functionality.

Can you share some of these uses cases, and privacy preserving alternatives you considered?  Unfortunately, "protect privacy via specifying suitable distribution mechanisms" is not an effective privacy-preservation strategy.

**Comment by sebastiankb** [comment]
Right now we have the TD call and we had a look into this issue. We would like to propose to have a common web call for next week to clarify all open questions. Is it possible for you to have a call next week Tuesday 23rd or Thursday 25th (8am PT)?

**Comment by takuki** [comment]
FYI,
I summarized the privacy risks and mitigations described in [section 9](https://w3c.github.io/wot-thing-description/#sec-security-consideration) in a few slides [here](https://github.com/w3c/wot-thing-description/files/3413082/TD.Privacy.pdf).

**Comment by samuelweiler** [comment]
> Is it possible for you to have a web meeting next week Tuesday 23rd or Thursday 25th (8am PT)?

Not for me.  That is the week of the IETF meeting, and adding remote calls to an IETF meeting week is fraught.

**Comment by sebastiankb** [comment]
@samuelweiler 
>Not for me. That is the week of the IETF meeting, and adding remote calls to an IETF meeting week is fraught. 

Well, if you are also attending the IETF meeting, there might be a chance that you can meet @mmccool there.

**Comment by mmccool** [comment]
Yes, let's have a meeting in person then.  Where and when are you @samuelweiler?  I will send an email... I will be at the T2TRG meetings today (Wed).

**Comment by snyderp** [comment]
I will be around and could meet Thursday AM before my flight if that would be useful @mmccool @samuelweiler.  8:30 / 9?

**Comment by sebastiankb** [comment]
quick update: 
id field is now optional and 'uniqueness' was removed from definition.
https://w3c.github.io/wot-thing-description/#thing

Please review.

**Comment by mmccool** [comment]
I have also updated the privacy risks and mitigations related to IDs (PR https://github.com/w3c/wot-thing-description/pull/825).  In particular, I added a "globally unique ID" risk (which explains why IDs do NOT need to be globally unique) and added a number of mitigations that take advantage of the fact that IDs are now optional (e.g. filtering out IDs when they are not needed for the use case).

This updated section is still informative, however.
